test/validation/org/apache/cassandra/HarnessTest.java

```diff
@@ -151,6 +153,11 @@ public void runModuleGroup(ArrayList<Module> modules)
             {
                 future.get();
             }
+
+            for (Future future : module_exceptions)
+            {
+                future.get();
```

Comment:
This is orthogonal to this PR, and can be done here, or in a future one, but since we're now repeating the block of code
```
for each future in a group of futures:
block until they're all done
```
Can you write a function that takes a collection of futures and handles this?